jerry-core/parser/js/opcodes-dumper.cpp

```diff
@@ -146,7 +253,10 @@ jsp_alloc_reg_for_temp (void)
 {
   JERRY_ASSERT (jsp_reg_max_for_local_var == VM_IDX_EMPTY);
 
-  vm_idx_t next_reg = jsp_reg_next++;
+  vm_idx_t next_reg = find_free_reg ();
+  set_reg_status (next_reg, REG_OCCUPIED);
```

Comment:
The call to the `find_free_reg` method already marks the register as used. So either we make the `find_free_reg` to not mark the register as used, or remove the `set_reg_status` call from here.